lib/commands/web.js

```diff
@@ -266,7 +266,13 @@ extensions.translateWebCoords = async function translateWebCoords (coords) {
   }
 };
 
-extensions.checkForAlert = async function checkForAlert () { // eslint-disable-line require-await
+extensions.checkForAlert = async function checkForAlert () {
+  try {
+    await this.getAlert();
```

Comment:
have you checked how this change affects the performance?

Reply:
maybe we could make this check configurable via settings, so it does not slow down tests, which don't face such issue

Reply:
There is very little scope for performance effects. Most of the atoms, which do not cause an alert to come up, return well within the 500ms timeout before we check the first alert. If they take longer, it will be extended a very small amount while the alert it checked.

In the case of alerts actually arising, this will be much faster than the current execution, which hangs until the operation times out.

I did a little check, with the `'element handling'` tests from the `appium-xcuitest-driver` Safari suite, which consists of 16 tests that involve running a lot of atoms. Running 5 times, the old strategy took `79767ms` on average, and this took `80591ms`.

If performance problems do arise, I'm happy to figure out something. But I don't think it is warranted at the moment.

Reply:
cool, thanks for checking it